include/mbgl/style/conversion_impl.hpp

```diff
@@ -286,6 +295,76 @@ optional<T> convert(const Convertible& value, Error& error, Args&&...args) {
     return Converter<T>()(value, error, std::forward<Args>(args)...);
 }
 
+template <typename T, typename = void>
+struct ValueFactory;
+
+template <typename T>
+struct ValueArrayFactory {
+    static Value make(const T& arg) { return mapbox::base::ValueArray(arg.begin(), arg.end()); }
+};
+
+template <>
+struct ValueFactory<std::array<float, 2>> : public ValueArrayFactory<std::array<float, 2>> {};
+
+template <>
+struct ValueFactory<std::vector<float>> : public ValueArrayFactory<std::vector<float>> {};
+
+template <>
+struct ValueFactory<ColorRampPropertyValue> {
+    static Value make(const ColorRampPropertyValue& value) { return value.getExpression().serialize(); }
+};
+
+template <>
+struct ValueFactory<TransitionOptions> {
+    static Value make(const TransitionOptions& value) {
+        return mapbox::base::ValueArray{
```

Comment:
I wonder if this could be an unordered map instead, with e.g. a missing key representing an empty optional for e.g. `duration` or `delay` values.

Reply:
I thought of it initially, but the current SDK implementation defaults those to `0` anyway. On the other hand vector is much lighter than unordered map.